File: sea-shell-service-immutable/src/main/java/org/jesperancinha/shell/webflux/immutable/data/SeaShellTopDto.java
package org.jesperancinha.shell.webflux.immutable.data;

import lombok.Builder;
import org.jesperancinha.shell.client.tops.Top;

public record SeaShellTopDto(
        String type,
        String color,
        String size
) {
    @Builder
    public SeaShellTopDto {

    }

    public static SeaShellTopDto create(Top top) {
        return SeaShellTopDto.builder()
                .type(top.getType())
                .color(top.getColor())
                .size(top.getSize())
                .build();
    }
}

File: sea-shell-service-immutable/src/main/kotlin/org/jesperancinha/shell/webflux/immutable/data/SeaShellTopDto.kt
package org.jesperancinha.shell.webflux.immutable.data

import org.jesperancinha.shell.client.tops.Top

data class SeaShellTopDto(val type: String, val color: String, val size: String) {
    companion object {
        fun create(top: Top?): SeaShellTopDto {
            return SeaShellTopDto(
                type = top!!.type,
                color = top.color,
                size = top.size
            )
        }
    }
}